test/unit/test_tone_analyzer_v3.py

```diff
@@ -1,145 +1,258 @@
+
+    def call_service(self, body):
+        service = ToneAnalyzerV3(
+            authenticator=NoAuthAuthenticator(),
+            version='2017-09-21',
+        )
+        service.set_service_url(base_url)
+        output = service.tone(**body)
+        return output
+
+    def construct_full_body(self):
+        body = dict()
+        body.update({"tone_input": {"mock": "data"}})
+        body['content_type'] = "string1"
+        body['sentences'] = True
+        body['tones'] = []
+        body['content_language'] = "string1"
+        body['accept_language'] = "string1"
+        return body
+
+    def construct_required_body(self):
+        body = dict()
+        body.update({"tone_input": {"mock": "data"}})
+        return body
+
+
+#-----------------------------------------------------------------------------
+# Test Class for tone_chat
+#-----------------------------------------------------------------------------
+class TestToneChat():
+
+    #--------------------------------------------------------
+    # Test 1: Send fake data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_tone_chat_response(self):
+        body = self.construct_full_body()
+        response = fake_response_UtteranceAnalyses_json
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 2: Send only required fake data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_tone_chat_required_response(self):
+        # Check response with required params
+        body = self.construct_required_body()
+        response = fake_response_UtteranceAnalyses_json
+        send_request(self, body, response)
         assert len(responses.calls) == 1
-        assert isinstance(ex, ApiException)
-        assert ex.code == error_code
-        assert ex.message == error_message
+
+    #--------------------------------------------------------
+    # Test 3: Send empty data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_tone_chat_empty(self):
+        check_empty_required_params(self, fake_response_UtteranceAnalyses_json)
+        check_missing_required_params(self)
+        assert len(responses.calls) == 0
+
+    #-----------
+    #- Helpers -
+    #-----------
+    def make_url(self, body):
+        endpoint = '/v3/tone_chat'
+        url = '{0}{1}'.format(base_url, endpoint)
+        return url
+
+    def add_mock_response(self, url, response):
+        responses.add(responses.POST,
+                      url,
+                      body=json.dumps(response),
+                      status=200,
+                      content_type='application/json')
+
+    def call_service(self, body):
+        service = ToneAnalyzerV3(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['ToneAnalyzerV3' may be undefined, or defined from star imports: ibm_watson.tone_analyzer_v3](https://app.codacy.com/manual/watson-developer-cloud/python-sdk/pullRequest?prid=4835642)